### PR TITLE
Client Runs page, detecting bad page selection

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?
This change allows the client runs page to revert back to the first page of nodes if the page selected does not have nodes. 

This problem was found with the IAM v2 global project filtering feature. When the project filter is updated it refreshes the page. Say the user is on page 2 with 100 nodes per page. If the user updates the project filter to have only 50 nodes then page 2 would be empty. Since the page 2 selection is in the URL after the refresh the user would be left with zero nodes. 

### :chains: Related Resources
#1341 
### :+1: Definition of Done

The repro steps in issue #1341 are resolved. 

### :athletic_shoe: How to Build and Test the Change
1. Build the changed code and start Automate: `build components/automate-ui && start_all_services`
1. Turn on IAM v2: `chef-automate iam upgrade-to-v2 --beta2.1`
1. Add some nodes to full at least two pages `chef_load_nodes 120`
1. Find the environment of one of less than 100 nodes. I found this from the search bar. 
1. Create and apply a project with the rule of the environment found above. 
1. Go to the client runs page https://a2-dev.test/infrastructure/client-runs.
1. Go to the second page of nodes. 
1. Apply the newly created project filter. 
1. Ensure that the client runs page goes back to the first page. 

### :white_check_mark: Checklist

- [x] Tests added/updated?
- [ ] Docs added/updated?

### :camera: Screenshots, if applicable
project_filter_client_runs

### CODE REVIEW

**Comment by phiggins** [comment]
Would it make sense to always go back to the first page if I change the project filter? Maybe I don't have the right model in my head, but I feel like if I change the project filter it's like I'm doing a new search and not a continuation of my existing one.

**Comment by lancewf** [comment]
@phiggins Yeah, I think that does make sense. That is what we do for all the other filter selections on the client runs page. For the project filter, we do not have an observer to when those changes are made. The way the project filter select changes are currently working is it refreshes the page.

File: components/automate-ui/src/app/pages/client-runs/client-runs.component.ts [hunk 1]
```diff
@@ -555,6 +574,15 @@ export class ClientRunsComponent implements OnInit, OnDestroy {
     this.store.dispatch(new UpdateNodeFilters({filters: nodeFilters}));
   }
 
+  isCurrentPageEmpty(nodes: Node[], loadedStatus: EntityStatus): boolean {
+    return loadedStatus === EntityStatus.loadingSuccess && nodes.length === 0;
+  }
+
+  // If there is a page parameter in the URL the first page is not selected.
+  isNotFirstPage(allUrlParameters: Chicklet[]): boolean {
+    return allUrlParameters.findIndex((parm: Chicklet) => parm.type === 'page') >= 0;
```
Comment (srenatus) [inline components/automate-ui/src/app/pages/client-runs/client-runs.component.ts]: [nit] How about [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)? It's pretty much the same except that it's already a boolean:
```suggestion
    return allUrlParameters.some((parm: Chicklet) => parm.type === 'page');
```

Comment (lancewf) [reply components/automate-ui/src/app/pages/client-runs/client-runs.component.ts]: Thank you! 
I always forget about 'some'. I wish there was an 'exists'.

File: components/automate-ui/src/app/pages/client-runs/client-runs.component.ts [hunk 2]
```diff
@@ -580,7 +580,7 @@ export class ClientRunsComponent implements OnInit, OnDestroy {
 
   // If there is a page parameter in the URL the first page is not selected.
   isNotFirstPage(allUrlParameters: Chicklet[]): boolean {
-    return allUrlParameters.findIndex((parm: Chicklet) => parm.type === 'page') >= 0;
+    return allUrlParameters.some((parm: Chicklet) => parm.type === 'page');
```
Comment (srenatus) [inline components/automate-ui/src/app/pages/client-runs/client-runs.component.ts]: [nit] `parm`? interesting choice. But it's OK, of course.

Comment (lancewf) [reply components/automate-ui/src/app/pages/client-runs/client-runs.component.ts]: Oh, I meant to have 'param', but I changed it to have the full 'parameter'.

**Pull request merged**